### PR TITLE
SR2024 - Update Section - Arduino API

### PR DESCRIPTION
**Pull request by 8BitJosh**
Update the Arduino API section to be more clear and update custom arduino section to match the new implimentation

### CODE REVIEW

File: programming/arduino/index.md [hunk 1]
```diff
@@ -3,134 +3,21 @@ layout: page
...
 ---
 
-Arduino API
-===========
+# Arduino API
 
-The [Ruggeduino](https://web.archive.org/web/20170317171649/https://www.rugged-circuits.com/ruggeduino)
-provides a total of 18 pins for either digital input or output (labelled 2 to 13 and A0 to A5),
-including 6 for analogue input (labelled A0 to A5).
+The Arduino that came as part of your kit was shipped with a firmware that provides the functionality outlined in the [Arduino API](/docs/programming/arduino/sr_firmware) page.
```
Comment (RealOrangeOne) [inline programming/arduino/index.md:8]: Suggestion: We should explicitly note here that using other arduinos is entirely supported (some may assume ours is _magic_)

File: programming/arduino/sr_firmware.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+robot.arduino.pins[4].mode = INPUT_PULLUP
+~~~~~
+
+
+## Input
+
+You can read a **digital** input pin with the following code:
+
+~~~~~ python
+# robot.arduinos[ARDUINO_SERIAL_NUMBER].pins[PIN_NO].digital_read()
```
Comment (RealOrangeOne) [inline programming/arduino/sr_firmware.md:80]: Question: Commented out code?

File: programming/arduino/sr_firmware.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+# to read Arduino's digital pin 3...
+pin3 = robot.arduino.pins[3].digital_read()
+~~~~~
+
+`pin3` will now contain `True` or `False` depending on whether the pin was high (5v) or low (0v), respectively.
+
+You can read an **analog** input pin with the following code:
+
+~~~~~ python
+# robot.arduinos[ARDUINO_SERIAL_NUMBER].pins[PIN_NO].analog_read()
```
Comment (RealOrangeOne) [inline programming/arduino/sr_firmware.md:91]: Question: Commented out code?

File: programming/arduino/sr_firmware.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+The analog pin numbers are available as `A0`, `A1`, `A2`, `A3`, `A4`, and `A5` respectively.
+
+
+## Output
+
+You can only set digital outputs (there's no analog_write, although feel free to modify the Arduino's firmware to add the ability to output [PWM](https://wikipedia.org/wiki/Pulse-width_modulation "Pulse-width modulation") if you desire).
+To set a digital output pin, you would use the following:
+
+~~~~~ python
+# robot.arduinos[ARDUINO_SERIAL_NUMBER].pins[PIN_NO].digital_write(VALUE)
```
Comment (RealOrangeOne) [inline programming/arduino/sr_firmware.md:106]: Question: Commented out code?

File: programming/arduino/sr_firmware.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+# to read Arduino's analog pin A0...
+pinA0 = robot.arduino.pins[A0].analog_read()
+~~~~~
+
+The analog pin numbers are available as `A0`, `A1`, `A2`, `A3`, `A4`, and `A5` respectively.
+
+
+## Output
+
+You can only set digital outputs (there's no analog_write, although feel free to modify the Arduino's firmware to add the ability to output [PWM](https://wikipedia.org/wiki/Pulse-width_modulation "Pulse-width modulation") if you desire).
```
Comment (RealOrangeOne) [inline programming/arduino/sr_firmware.md:102]:
```suggestion
You can only set digital outputs (there's no analog_write, although feel free to modify the Arduino's firmware to add the ability to output [PWM](https://wikipedia.org/wiki/Pulse-width_modulation) ("Pulse-width modulation") if you desire).
```

Comment (8BitJosh) [reply programming/arduino/sr_firmware.md]: 33a3875

File: programming/arduino/sr_firmware.md (hunk 4 above)
Comment (8BitJosh) [reply programming/arduino/sr_firmware.md]: 33a3875

File: programming/arduino/sr_firmware.md (hunk 3 above)
Comment (8BitJosh) [reply programming/arduino/sr_firmware.md]: 33a3875

File: programming/arduino/sr_firmware.md (hunk 2 above)
Comment (8BitJosh) [reply programming/arduino/sr_firmware.md]: 33a3875

File: programming/arduino/index.md (hunk 1 above)
Comment (8BitJosh) [reply programming/arduino/index.md:8]: 33a3875

File: programming/arduino/custom_firmware.md [hunk 6]
```diff
@@ -3,148 +3,140 @@ layout: page
...
-This is accessible from the `ignored_ruggeduinos` dictionary.
+### write
+
+`write` is used to send data to the serial port, the function will send whatever you provide.
+Putting a `b` in front of a string is a short hand way of creating a bytes object.
 
 ~~~~~ python
-ruggeduino_device = R.ignored_ruggeduinos[RUGGEDUINO_ID]
+# This will send the message "data" over the serial port
+robot.raw_serial_devices[ARDUINO_SN].write(b"data")
```
[the remark this answers is not in the document]
Comment (8BitJosh) [reply programming/arduino/custom_firmware.md:89]: 33a3875

File: programming/arduino/custom_firmware.md [hunk 7]
```diff
@@ -3,148 +3,140 @@ layout: page
...
-By default, the [`sr.robot3`](/docs/programming/robot_api/) library assumes that all connected Ruggeduinos are running the SR firmware
-or firmware which is compatible with the SR Ruggeduino firmware.
-If you're using completely custom firmware, you'll need to tell the kit to ignore the ruggeduino so that you're able to define your own setup logic.
+By default when the `robot` object is created it will try to communicate with all Arduinos and will expect them to respond in the way the SR firmware does.
+If you want the API to not try connecting to a device we need to *ignore* the Arduino.
 
-[Extension of the SR firmware](#extension) {#extension}
-------------------------------
+To configure a `Robot` object to ignore a Arduino with custom firmware, you will need to provide it with the Arduino's serial number.
+The Arduino serial number is a character string of numbers and letters, and is output in the robot log when you run a program on your robot with your
```
[the remark this answers is not in the document]
Comment (8BitJosh) [reply programming/arduino/custom_firmware.md]: 33a3875

File: programming/arduino/custom_firmware.md [hunk 8]
```diff
@@ -3,148 +3,140 @@ layout: page
...
-      case 'r':
-        command_read();
-        break;
-      case 'l':
-        command_write(LOW);
-        break;
-
- // ... and so on ...
-~~~~~
+You'll need the ID later, so it's best to save it into a variable:
```
[the remark this answers is not in the document]
Comment (8BitJosh) [reply programming/arduino/custom_firmware.md]: 33a3875

**Review by WillB97 — APPROVED** [review]

**Pull request merged**